core/controllers/reader.py

```diff
@@ -294,6 +298,69 @@ def get(self, exploration_id):
         self.render_json(self.values)
 
 
+class StorePlaythroughHandler(base.BaseHandler):
+    """Handles a useful playthrough coming in from the frontend to store it."""
+
+    @acl_decorators.can_play_exploration
+    def post(self, exploration_id):
+        """Handles POST requests. Appends to existing list of playthroughs or
+        deletes it if already full.
+
+        Args:
+            exploration_id: str. The ID of the exploration.
+        """
+        exp_issue_dict = self.payload.get('exp_issue')
+        playthrough_data = self.payload.get('playthrough_data')
+
+        exp_issues_model = stats_models.ExplorationIssuesModel.get(
+            exploration_id)
+        exp_issues = stats_services.get_exp_issues_from_model(exp_issues_model)
+
+        playthrough_id = stats_models.PlaythroughModel.create(
+            playthrough_data['exp_id'],
+            playthrough_data['exp_version'],
+            playthrough_data['issue_type'],
+            playthrough_data['issue_customization_args'],
+            playthrough_data['playthrough_actions'],
+            playthrough_data['is_valid'])
+
+        customization_args = exp_issue_dict['customization_args']
+
+        issue_found = False
+        for index, issue in enumerate(exp_issues.unresolved_issues):
+            if issue['issue_type'] == exp_issue_dict['issue_type']:
+                issue_customization_args = issue['issue_customization_args']
+                # In case issue_keyname is 'state_names', the ordering of the
+                # list is important i.e. [a,b,c] is different from [b,c,a].
+                issue_keyname = feconf.ISSUE_TYPE_KEYNAME_MAPPING[issue[
+                    'issue_type']]
+                if (issue_customization_args[issue_keyname] ==
+                        customization_args[issue_keyname]):
+                    if (len(issue['playthrough_ids']) <
+                            feconf.MAX_PLAYTHROUGHS_FOR_ISSUE):
+                        exp_issues.unresolved_issues[index][
+                            'playthrough_ids'].append(playthrough_id)
+                    else:
+                        # In this case, the playthrough instance created doesn't
+                        # need to be stored, so it is deleted.
+                        stats_models.PlaythroughModel.get(
+                            playthrough_id).delete()
+                    issue_found = True
+                    break
+
+        if not issue_found:
+            issue = {
+                'issue_type': playthrough_data['issue_type'],
+                'issue_customization_args': playthrough_data[
+                    'issue_customization_args'],
+                'playthrough_ids': [playthrough_id]
+            }
+            exp_issues.unresolved_issues.append(issue)
```

Comment:
Any problem if there's already a resolved issue with the same issue type and customization args? Will it lead to duplicates when the unresolved issue subsequently gets resolved?

Reply:
@seanlip Our current system of handling issue resolving is to just remove the issue from the list and delete all associated playthroughs. So, if an issue has the same type and customization args and we've resolved an issue before of the same type, the new issue will still get recorded. This behaviour is desired since the issue is cropping up even after the creator resolves an issue.

Reply:
OK, sounds good, thanks!